ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java

```diff
@@ -0,0 +1,756 @@
+                    checkFilter(name);
+                }
+                return super.resolveProxyClass(unmarshaller, interfaces);
+            }
+
+            private void checkFilter(String className) throws InvalidClassException {
+                if (classResolverFilter != null && classResolverFilter.apply(className) != Boolean.TRUE) {
+                    throw EjbHttpClientMessages.MESSAGES.cannotResolveFilteredClass(className);
+                }
+
+            }
+        }
+    }
+
+    private static final class HttpCancelHandler extends RemoteHTTPHandler {
+
+        private final Map<InvocationIdentifier, CancelHandle> cancellationFlags;
+
+        HttpCancelHandler(ExecutorService executorService, Map<InvocationIdentifier, CancelHandle> cancellationFlags) {
+            super(executorService);
+            this.cancellationFlags = cancellationFlags;
+        }
+
+        @Override
+        protected void handleInternal(HttpServerExchange exchange) throws Exception {
+            String ct = exchange.getRequestHeaders().getFirst(Headers.CONTENT_TYPE);
+            ContentType contentType = ContentType.parse(ct);
+            if (contentType != null) {
+                exchange.setStatusCode(StatusCodes.BAD_REQUEST);
+                EjbHttpClientMessages.MESSAGES.debugf("Bad content type %s", ct);
+                return;
+            }
+
+            String relativePath = exchange.getRelativePath();
+            if (relativePath.startsWith("/")) {
+                relativePath = relativePath.substring(1);
+            }
+            String[] parts = relativePath.split("/");
+            if (parts.length != 6) {
+                exchange.setStatusCode(StatusCodes.NOT_FOUND);
+                return;
+            }
+            final String app = handleDash(parts[0]);
+            final String module = handleDash(parts[1]);
+            final String distinct = handleDash(parts[2]);
+            final String bean = parts[3];
+            String invocationId = parts[4];
+            boolean cancelIdRunning = Boolean.parseBoolean(parts[5]);
+            Cookie cookie = exchange.getRequestCookie(JSESSIONID_COOKIE_NAME);
+            final String sessionAffinity = cookie != null ? cookie.getValue() : null;
+            final InvocationIdentifier identifier;
+            if (invocationId != null && sessionAffinity != null) {
+                identifier = new InvocationIdentifier(invocationId, sessionAffinity);
+            } else {
+                exchange.setStatusCode(StatusCodes.BAD_REQUEST);
+                EjbHttpClientMessages.MESSAGES.debugf("Exchange %s did not include both session id and invocation id in cancel request", exchange);
+                return;
+            }
+            CancelHandle handle = cancellationFlags.remove(identifier);
+            if (handle != null) {
+                handle.cancel(cancelIdRunning);
+            }
+        }
+
+        private static String handleDash(String s) {
```

Comment:
Move handleDash to RemoteHTTPHandler as a common method.

Reply:
Done.